iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -171,6 +172,16 @@ public void preCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable)
     hmsTable.getParameters().put(BaseMetastoreTableOperations.TABLE_TYPE_PROP,
         BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toUpperCase());
 
+    // Make sure the created iceberg table location on external warehouse location
+    if (Catalogs.hiveCatalog(conf, catalogProperties) && hmsTable.getSd().getLocation() == null &&
+        !TableType.EXTERNAL_TABLE.equals(hmsTable.getTableType())) {
+      Map<String, String> params = Optional.ofNullable(hmsTable.getParameters()).orElse(Maps.newHashMap());
+      params.put("EXTERNAL", "TRUE");
+      params.put(MetaStoreUtils.EXTERNAL_TABLE_PURGE, "TRUE");
+      hmsTable.setParameters(params);
+      hmsTable.setTableType(TableType.EXTERNAL_TABLE.toString());
```

Comment:
@ayushtkn 
The code block is responsible for transforming created iceberg table which has no `external` keyword  to external table, and then the iceberg location will always go to external WH location. 

The change is in line with `MetastoreDefaultTransformer::transformCreateTable`:
https://github.com/apache/hive/blob/7bfc54ffc8baf5f3e7de326b750fe9355b11132b/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreDefaultTransformer.java#L651-L658

Reply:
http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-4341/8/tests/
Changing the type of the table to `EXTERNAL ` seems to affect the execution plan. Not sure why this is happening.

I determine to use a easy workaroud that setting the location to external WH location explicitly in `HiveIcebergMetaHook::preCreateTable`:

Reply:
I think put external only when the transformer is null, else let it go its own way.
Some output were changing, like the table was managed and now external, I think for those we can regenerate the output

Reply:
> I think put external only when the transformer is null, else let it go its own way. Some output were changing, like the table was managed and now external, I think for those we can regenerate the output

@ayushtkn That's not true. I have found out this issue that `EXTERNAL ` iceberg table can not use statistics to optimize query, and so the physical execution plan was changed. I have filed a PR https://github.com/apache/hive/pull/4388, please take a look. Also cc @deniskuzZ